From 2e9b0acf383bf5bc8881d563a58deaf3da13602f Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Tue, 21 Oct 2025 15:01:44 +0200
Subject: [PATCH 463/553] CMake: Add more compiler / linker identification info

Useful when builders share logs.

Pick-to: 6.8
Change-Id: I0091203f906aa5b43384c474f42fa555d1e86dba
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
(cherry picked from commit b705d42071f6abe1d949c4975a8d2fe4da9dee52)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtBaseConfigureTests.cmake | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/cmake/QtBaseConfigureTests.cmake b/cmake/QtBaseConfigureTests.cmake
index c50723e43a2..ade53369b46 100644
--- a/cmake/QtBaseConfigureTests.cmake
+++ b/cmake/QtBaseConfigureTests.cmake
@@ -306,6 +306,36 @@ function(qt_internal_print_cmake_host_and_target_info)
     message(STATUS "CMAKE_SYSTEM_PROCESSOR: \"${CMAKE_SYSTEM_PROCESSOR}\"")
 
     message(STATUS "CMAKE_CROSSCOMPILING: \"${CMAKE_CROSSCOMPILING}\"")
+
+    message(STATUS "CMAKE_CXX_COMPILER_ID: \"${CMAKE_CXX_COMPILER_ID}\"")
+    message(STATUS "CMAKE_CXX_COMPILER_VERSION: \"${CMAKE_CXX_COMPILER_VERSION}\"")
+
+    # The variables might not be defined depending on platform and CMake version.
+    if(CMAKE_CXX_COMPILER_TARGET)
+        message(STATUS "CMAKE_CXX_COMPILER_TARGET: \"${CMAKE_CXX_COMPILER_TARGET}\"")
+    endif()
+    if(CMAKE_CXX_COMPILER_ARCHITECTURE_ID)
+        message(STATUS
+            "CMAKE_CXX_COMPILER_ARCHITECTURE_ID: \"${CMAKE_CXX_COMPILER_ARCHITECTURE_ID}\"")
+    endif()
+    if(CMAKE_CXX_COMPILER_FRONTEND_VARIANT)
+        message(STATUS
+            "CMAKE_CXX_COMPILER_FRONTEND_VARIANT: \"${CMAKE_CXX_COMPILER_FRONTEND_VARIANT}\"")
+    endif()
+
+    if(CMAKE_CXX_COMPILER_LINKER_ID)
+        message(STATUS "CMAKE_CXX_COMPILER_LINKER_ID: \"${CMAKE_CXX_COMPILER_LINKER_ID}\"")
+    endif()
+
+    if(CMAKE_CXX_COMPILER_LINKER_VERSION)
+        message(STATUS
+            "CMAKE_CXX_COMPILER_LINKER_VERSION: \"${CMAKE_CXX_COMPILER_LINKER_VERSION}\"")
+    endif()
+
+    if(CMAKE_CXX_COMPILER_LINKER_FRONTEND_VARIANT)
+        message(STATUS "CMAKE_CXX_COMPILER_LINKER_FRONTEND_VARIANT: "
+            "\"${CMAKE_CXX_COMPILER_LINKER_FRONTEND_VARIANT}\"")
+    endif()
 endfunction()
 qt_internal_print_cmake_host_and_target_info()
 
-- 
2.51.2

